lib/engine/game/base.rb

```diff
@@ -1261,11 +1265,20 @@ def priority_deal_player
         end
       end
 
-      def reorder_players
-        player = @players.reject(&:bankrupt)[@round.entity_index]
+      def next_sr_position(entity)
+        player_order = @round.current_entity&.player? ? @round.pass_order : @players
+        player_order.reject(&:bankrupt).index(entity)
+      end
 
-        @players.rotate!(@players.index(player))
-        @log << "#{player.name} has priority deal"
+      def reorder_players
+        case self.class::NEXT_SR_PLAYER_ORDER
+        when :after_last_to_act
+          player = @players.reject(&:bankrupt)[@round.entity_index]
+          @players.rotate!(@players.index(player))
```

Comment:
should everything use pass order instead of last to pass?

Reply:
Pass order is a bit different. Its tracking when players first started passing, not just the last round of passes. For example, if the SR actions were like:

P1 buy, pass
P2 pass, pass
P3 buy, pass

The pass order would be P2, P1, P3; not P1, P2, P3.